python/add_corona.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+
+#===================================================
+#
+#    Copyright (c) 2025
+#      SMASH Hybrid Team
+#
+#    GNU General Public License (GPLv3 or later)
+#
+#===================================================
+
+import argparse
+import numpy as np
+import sys
+import os
+import subprocess
+
+'''
+    This script adds the corona particles from the initial SMASH run
+    to the sampled particle list.
+'''
+
+def find_out_event_number(filepath):
+    '''
+        Look into the last line of the file and find the number of events
+    '''
+    if not os.path.isfile(filepath):
+        print(filepath+" does not exist!")
+        sys.exit(3)
+    tail = subprocess.run(['tail', '-n', '1', filepath],
+                          stdout=subprocess.PIPE,stderr=subprocess.PIPE)
+    last_line = tail.stdout.decode().strip()
+    if "end" in last_line:
```

Comment:
Since this was exactly a bug in SMASH recently, I'd be in favor of
```suggestion
    if " end " in last_line:
```
which makes it slightly safer.

Reply:
actually I changed the line to be `split()`, i.e. its an array that needs to contain the element `"end"`, instead of the full string

Reply:
I like it! :+1: